### PR TITLE
Renaming PhoneNumberSearch to PhoneNumberReservation and adding purchaseReservation LRO

### PR DESCRIPTION
**Pull request by paolamvhz**

### CODE REVIEW

File: sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java [hunk 1]
```diff
@@ -819,19 +819,82 @@
...
+            });
+
+    }
+
+
+    private Function<PollingContext<Void>,
+        Mono<Void>> purchaseReservationFetchResultOperation() {
+
+        return pollingContext -> {
+
```
Comment (jbeauregardb) [inline sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java:892]: nit: remove empty lines

**Comment by jbeauregardb** [comment]
LGTM 👍🏻 Thank you for renaming the model. The only thing missing would be changing the README to make sure it also has the renamed model documented.

File: sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java [hunk 2]
```diff
@@ -819,19 +819,82 @@
...
+        return (pollingContext) -> {
+            Mono<Void> response = purchaseSearchWithResponse(reservationId).map(
+                purchaseResponse -> purchaseResponse.getValue());
+            return response;
+        };
+    }
+
+    private Function<PollingContext<Void>, Mono<PollResponse<Void>>>
+                purchaseReservationCreatePollOperation(String reservationId) {
```
Comment (ankitarorabit) [inline sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java:868]: nit: shouldn't this be purchaseReservationPollOperation (don't understand the 'create' in between)

File: sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java [hunk 3]
```diff
@@ -819,19 +819,82 @@
...
+
+    }
+
+
+    private Function<PollingContext<Void>,
+        Mono<Void>> purchaseReservationFetchResultOperation() {
+
+        return pollingContext -> {
+
+            return Mono.just(pollingContext.getLatestResponse().getValue());
```
Comment (ankitarorabit) [inline sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java:893]: What does the value contain? Isnt that the 'SearchStatus', in that case shouldnt you change the return type of this method and for the flux (its currently void)?

**Review by ankitarorabit — CHANGES_REQUESTED** [review]
Added comments.

File: sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java (hunk 3 above)
Comment (jbeauregardb) [reply sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java:893]: She's returning a null in the pollOperation so this is fine, I guess

Comment (ankitarorabit) [reply sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java:893]: ohh yea, you are right 👍

**Comment by paolamvhz** [comment]
The changes where applied in other PR

**Pull request closed without merging**